test/nodetool/test_nodetool.py

```diff
@@ -0,0 +1,89 @@
+#
+# Copyright 2023-present ScyllaDB
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+#
+
+import pytest
+import rest_api_mock
+import json
+import subprocess
+
+
+def test_jmx_compatibility_args(server_context, nodetool):
+    """Test that JMX related command line arguments are ignored.
+
+    Since we cannot invoke nodetool without a concrete command, we use nodetool
+    compact here, but this is irrelevant, we are just checking that JMX-related
+    command-line args are recognized and ignored.
+    """
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema", "-u", "us3r", "-pw", "secr3t")
+    nodetool("compact", "system_schema", "--username", "us3r", "--password", "secr3t")
+    nodetool("compact", "system_schema", "-u", "us3r", "-pwf", "/tmp/secr3t_file")
+    nodetool("compact", "system_schema", "--username", "us3r", "--password-file", "/tmp/secr3t_file")
+    nodetool("compact", "system_schema", "-pp")
+    nodetool("compact", "system_schema", "--print-port")
+
+
+def test_compact_all_keyspaces(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("GET", "/storage_service/keyspaces?type=all", json.dumps(["system", "system_schema"])),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system"),
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact")
+
+
+def test_compact_keyspace(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema")])
+
+    nodetool("compact", "system_schema")
+
+
+def test_compact_table(server_context, nodetool):
+    rest_api_mock.set_expected_requests(server_context, [
+        rest_api_mock.request("POST", "/storage_service/keyspace_compaction/system_schema?cf=columns")])
+
+    nodetool("compact", "system_schema", "columns")
```

Comment:
Is it possible to list several tables? If so, please test that too.

Reply:
Yes, will do.

Reply:
Done in v4.